File: 3rd Sem/NM/C-program/guassianiteration.c
#include <stdio.h>
#include <math.h>

double f(double x) {
    return x * x;  // Function to be integrated, x^2 in this case
}

double gauss_integration(int n, double a, double b) {
    double x[2] = {-1 / sqrt(3), 1 / sqrt(3)};
    double w[2] = {1, 1};
    
    if (n != 2) {
        printf("Only n=2 is supported here\n");
        return -1;
    }
    
    double c1 = (b - a) / 2;
    double c2 = (b + a) / 2;
    double integral = 0;
    
    for (int i = 0; i < n; i++) {
        integral += w[i] * f(c1 * x[i] + c2);
    }
    
    return c1 * integral;
}

int main() {
    double a, b;
    
    printf("Enter lower and upper limits: ");
    scanf("%lf %lf", &a, &b);
    
    int n = 2;
    double result = gauss_integration(n, a, b);
    
    printf("Integral: %.6f\n", result);
    
    return 0;
}

File: 3rd Sem/NM/python/guassianiteration.py
import math

def f(x): 
    return x ** 2

def gauss_integration(n, a, b):
    if n == 2:
        x = [-1 / math.sqrt(3), 1 / math.sqrt(3)]
        w = [1, 1]
    else:
        raise ValueError("Only n=2 supported here")

    c1 = (b - a) / 2
    c2 = (b + a) / 2
    integral = 0
    for i in range(n):
        integral += w[i] * f(c1 * x[i] + c2)

    return c1 * integral

# Example usage
a, b = 0, 8  # Integration limits
n = 2         # Number of points for Gaussian quadrature (must be 2 here)
result = gauss_integration(n, a, b)
print("Integral:", result)
